spring-web/src/main/java/com/webcohesion/enunciate/modules/spring_web/EnunciateSpringWebContext.java

```diff
@@ -145,8 +145,14 @@ public List<ResourceGroup> getResourceGroupsByClass(ApiRegistrationContext regis
     List<ResourceGroup> resourceGroups = new ArrayList<ResourceGroup>();
     Set<String> slugs = new TreeSet<String>();
     FacetFilter facetFilter = registrationContext.getFacetFilter();
+
     for (SpringController springController : controllers) {
-      if (!facetFilter.accept(springController)) {
+      // NOTE: do not call facetFilter.accept(springController) to check if evaluating RequestMapping's in the controller can be skipped
+      // - If an "included" facet is defined, and not applied to the controller; then checking facetFilter.accept(springController) will
+      //   return false even if the included facet is applied to a RequestMapping inside the controller
+      // - RequestMappings inherit all the facets applied directly to the controller, so skipping the check on the controller won't cause
+      //   problems with included/excluded facets that are applied to the controller
+      if (springController.getRequestMappings().stream().noneMatch(facetFilter::accept)) {
         continue;
       }
```

Comment:
`getResourceGroupsByClass()` is the way that enunciate is set up in my spring-boot app. I tested these changes in my app and they work as I expect them to.

I am not sure how to configure enunciate to use `getResourceGroupsByPath()` or `getResourceGroupsByAnnotation()`; so I have not tested the changes I made to those code paths.

The JarsX module also had code that looked very similar to the SpringWeb module. I applied the same code-changes there that I did hers, but I don't have any JarsX projects to test the changes with

Reply:
getResourceGroupsByClass uses a `getRequestMappings().stream().noneMatch()` precheck so that the `slugs.add(slug);` line is not hit when there are no requestMappings included. This does mean that we will loop through the requestMappings in the controller twice instead of once.

I don't think that this will be a performance problem. If performance is a problem we could
- remove the pre-check and call `slugs.remove(slug);` at the end of the loop if `group.getResources().isEmpty()`
- add a new method to facetFilter that only checks excluded facets and add before the requestMappings precheck a call to this new method. `if (!facetFilter.hasExcludedFacet(springController)) { continue; }`

---

`getResourceGroupsByPath()` and `getResourceGroupsByAnnotation()` only loop through the requestMappings one time (no pre-check) because they do not modify any state before they start checking the requestMappings